packages/chakra-ui-core/src/CNumberInput/CNumberInput.js

```diff
@@ -436,7 +444,11 @@ const CNumberInput = {
         align: 'stretch',
         w: this.isFullWidth ? 'full' : null,
         pos: 'relative'
-      }
+      },
+      nativeOn: {
+        input: this.emitValue
+      },
+      ref: 'input'
```

Comment:
Hey @peoray ! Thanks for the PR. Have a few things to make some changes for.

Even though the event there is the native input coming from the `CNumberInput` element, it's actually not the target element associated with the event. So I don't think it's a good idea to listen to the native event here and rely on it. 

The better place to handle the emitted changes would be inside the `this.handleChange` method. We only need to emit the `change` event whenever a new value is entered into the `CNumberInputField`.

So technically the `v-model` binding should work by changing the `model` property to be:
```js
model: {
  event: 'change',
  prop: 'value'
}
```